openmetadata-ui/src/main/resources/ui/src/components/DataContract/ContractSemanticFormTab/ContractSemanticFormTab.tsx

```diff
@@ -203,9 +203,7 @@ export const ContractSemanticFormTab: React.FC<{
           <Form.List name="semantics">
             {(fields, { add }) => {
               // Store the add function so it can be used outside
-              if (!addFunctionRef.current) {
-                addFunctionRef.current = add;
-              }
+              addFunctionRef.current ??= add;
```

Comment:
You can add null check here to fix code smell instead removing conditino completly